src/main/java/frc/robot/commands/autonomous/PathAutoBase.java

```diff
@@ -0,0 +1,66 @@
+package frc.robot.commands.autonomous;
+
+import com.pathplanner.lib.auto.AutoBuilder;
+import com.pathplanner.lib.commands.PathPlannerAuto;
+import com.pathplanner.lib.path.PathPlannerPath;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.wpilibj.DriverStation;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.Commands;
+import frc.robot.RobotContainer;
+
+public class PathAutoBase extends PathPlannerAuto {
+  public PathAutoBase() {
+    super(Commands.run(() -> {}));
+  }
+
+  protected Pose2d getStartingPose(PathPlannerPath thePath) {
+    return new Pose2d(thePath.getPoint(0).position, thePath.getIdealStartingState().rotation());
+  }
+
+  protected PathAutoBase sequencePathThenCommand(
+      PathPlannerPath thePath, double scoringTime, Command then, String name) {
+    isRunning()
+        .onTrue(
+            Commands.sequence(
+                    AutoBuilder.resetOdom(getStartingPose(thePath)),
+                    AutoBuilder.followPath(thePath),
+                    Commands.waitSeconds(scoringTime),
+                    Commands.runOnce(() -> then.schedule()))
+                .withName(name));
+    return this;
+  }
+
+  protected static PathAutoBase sequencePathThenCommand(
+      String pathname, double scoringTime, Command then, String name) {
+    PathAutoBase pathAuto = new PathAutoBase();
+    try {
+      PathPlannerPath thePath = PathPlannerPath.fromPathFile(pathname);
+      return pathAuto.sequencePathThenCommand(thePath, scoringTime, then, name);
+    } catch (Exception e) {
+      DriverStation.reportError("Path Not Found: " + e.getMessage(), e.getStackTrace());
+    }
+    return pathAuto;
+  }
+
+  public static PathPlannerAuto fiveCycleNonProcessor(RobotContainer robotContainer) {
```

Comment:
@isaacngkh This should be a way to consolidate and share more of the auto startup code.
With this, we can delete 4 other classes.
For the part 2 commands, we should make a super-class maybe also?
We will want to put the shared logic for scoring the coral in some place too